app/user_test.go

```diff
@@ -653,6 +653,88 @@ func TestCreateUserWithToken(t *testing.T) {
 		require.Len(t, *members, 1)
 		assert.Equal(t, (*members)[0].ChannelId, th.BasicChannel.Id)
 	})
+
+	t.Run("create guest having email domain restrictions", func(t *testing.T) {
+		enableGuestDomainRestricions := *th.App.Config().GuestAccountsSettings.RestrictCreationToDomains
+		defer func() {
+			th.App.UpdateConfig(func(cfg *model.Config) {
+				cfg.GuestAccountsSettings.RestrictCreationToDomains = &enableGuestDomainRestricions
+			})
+		}()
+		th.App.UpdateConfig(func(cfg *model.Config) { *cfg.GuestAccountsSettings.RestrictCreationToDomains = "restricted.com" })
+		forbiddenInvitationEmail := model.NewId() + "other-email@test.com"
+		grantedInvitationEmail := model.NewId() + "other-email@restricted.com"
+		forbiddenDomainToken := model.NewToken(
+			TOKEN_TYPE_GUEST_INVITATION,
+			model.MapToJson(map[string]string{"teamId": th.BasicTeam.Id, "email": forbiddenInvitationEmail, "channels": th.BasicChannel.Id}),
+		)
+		grantedDomainToken := model.NewToken(
+			TOKEN_TYPE_GUEST_INVITATION,
+			model.MapToJson(map[string]string{"teamId": th.BasicTeam.Id, "email": grantedInvitationEmail, "channels": th.BasicChannel.Id}),
+		)
+		require.Nil(t, th.App.Srv.Store.Token().Save(forbiddenDomainToken))
+		require.Nil(t, th.App.Srv.Store.Token().Save(grantedDomainToken))
+		guest := model.User{
+			Email:       strings.ToLower(model.NewId()) + "+test@example.com",
+			Nickname:    "Darth Vader",
+			Username:    "vader" + model.NewId(),
+			Password:    "passwd1",
+			AuthService: "",
+		}
+		newGuest, err := th.App.CreateUserWithToken(&guest, forbiddenDomainToken)
+		require.NotNil(t, err)
+		require.Nil(t, newGuest)
+		require.Equal(t, "api.user.create_user.accepted_domain.app_error", err.Id)
```

Comment:
Same as above. This can be assert.Equal. If this fails, the entire test will be stopped, which is not what we want I believe.